process/block/sovereignChainBlock.go

```diff
@@ -829,14 +834,30 @@ func (scbp *sovereignChainBlockProcessor) checkExtendedShardHeadersValidity() er
 		return err
 	}
 
-	log.Trace("checkExtendedShardHeadersValidity", "lastCrossNotarizedHeader nonce", lastCrossNotarizedHeader.GetNonce())
 	extendedShardHdrs := scbp.sortExtendedShardHeadersForCurrentBlockByNonce()
 	if len(extendedShardHdrs) == 0 {
 		return nil
 	}
 
+	log.Trace("checkExtendedShardHeadersValidity",
+		"lastCrossNotarizedHeader nonce", lastCrossNotarizedHeader.GetNonce(),
+		"lastCrossNotarizedHeader round", lastCrossNotarizedHeader.GetRound(),
+	)
+	if scbp.receivedGenesisMainChainHeaderWithoutPreGenesis(extendedShardHdrs[0]) {
+		// we are missing pre-genesis header, so we can't link it to previous header
+		if len(extendedShardHdrs) == 1 {
+			return nil
+		}
+
+		lastCrossNotarizedHeader = extendedShardHdrs[0]
```

Comment:
The `extendedShardHdrs[0]` is not logged

Reply:
Right, we could log it as well.